### PR TITLE
Search (refactored pull request)

### PR DESCRIPTION
**Pull request by petzlux**
Left out the .po changes, that were introduced accidentally

### CODE REVIEW

**Comment by petzlux** [comment]
Please review this pull request

**Comment by elemoine** [comment]
Reviewing now.

**Comment by elemoine** [comment]
Where is `geoportailv3/utils/db_to_es.py` used? What is it for? A comment at the head of the file indicating what the file does and what uses it would be useful in my opinion.

File: geoportailv3/lib/search.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            'public': {'type': 'boolean', 'index': 'not_analyzed'},
+            'params': {'type': 'string', 'index': 'not_analyzed'},
+            'role_id': {'type': 'integer', 'index': 'not_analyzed'},
+            'ts': {'type': 'geo_shape'},
+        }
+    }
+}
+
+
+def get_es(request):
```
Comment (elemoine) [inline geoportailv3/lib/search.py]: I'd rather name this `get_eleasticsearch`, to make the code more explicit and readable.

File: geoportailv3/views/fulltextsearch.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            request.response.cache_control.private = True
+        request.response.cache_control.max_age = \
+            request.registry.settings["default_max_age"]
+        self.settings = request.registry.settings.get('fulltextsearch', {})
+
+    @view_config(route_name='fulltextsearch', renderer='geojson')
+    def fulltextsearch(self):
+        if 'query' not in self.request.params:
+            return HTTPBadRequest(detail='no query')
+        query = self.request.params.get('query')
```
Comment (elemoine) [inline geoportailv3/views/fulltextsearch.py]: Is `query` used?

File: geoportailv3/views/fulltextsearch.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            return HTTPBadRequest(detail='limit value is incorrect')
+        if limit > maxlimit:
+            limit = maxlimit
+
+        try:
+            partitionlimit = int(self.request.params.get('partitionlimit', 0))
+        except ValueError:
+            return HTTPBadRequest(detail='partitionlimit value is incorrect')
+        if partitionlimit > maxlimit:
+            partitionlimit = maxlimit
```
Comment (elemoine) [inline geoportailv3/views/fulltextsearch.py]: Is `partitionlimit` used? If not then it should be removed.

**Comment by elemoine** [comment]
I've added a couple comments. Otherwise looks good to me.

File: geoportailv3/views/fulltextsearch.py (hunk 2 above)
Comment (petzlux) [inline geoportailv3/views/fulltextsearch.py]: Yes, its the string to search for ...

File: geoportailv3/views/fulltextsearch.py (hunk 3 above)
Comment (petzlux) [inline geoportailv3/views/fulltextsearch.py]: This code is from c2cgeoportal, wanted to stick to your interface, but if its not used, than happy to remove it  ...

File: geoportailv3/views/fulltextsearch.py (hunk 2 above)
Comment (elemoine) [inline geoportailv3/views/fulltextsearch.py]: Oh sorry, I had missed it.

File: geoportailv3/views/fulltextsearch.py (hunk 3 above)
Comment (elemoine) [inline geoportailv3/views/fulltextsearch.py]: I don't know if the notion of partitions is applicable when using ElasticSearch.

File: geoportailv3/lib/search.py (hunk 1 above)
Comment (petzlux) [inline geoportailv3/lib/search.py]: Done

File: geoportailv3/views/fulltextsearch.py (hunk 3 above)
Comment (petzlux) [inline geoportailv3/views/fulltextsearch.py]: Okay, I will remove for now, as I cant see the point in it , until we integrate faceting (filtering by layer_name for example)

**Pull request merged**